WcaOnRails/db/migrate/20231122153755_create_wfc_dues_redirects.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class CreateWfcDuesRedirects < ActiveRecord::Migration[7.0]
+  def change
+    create_table :wfc_dues_redirects do |t|
+      t.string :redirect_type, null: false
+      t.references :redirect_from_country, type: :string, foreign_key: { to_table: :Countries }
+      t.references :redirect_from_organizer, type: :integer, foreign_key: { to_table: :users }
```

Comment:
Rails has a built-in way of handling polymorphic associations: https://guides.rubyonrails.org/association_basics.html#polymorphic-associations

You have to redesign your database model slightly, but it will give you a huge benefit in terms of usability and code convenience.

Reply:
```suggestion
      t.references :redirect_source, polymorphic: true
```